db/dataset_eval.py

```diff
@@ -143,6 +148,33 @@ def validate_dataset_contents(dataset):
                 )
 
 
+def filter_dataset_contents(dataset):
+    """Filter dataset contents by ignoring the recordings referenced
+       in classes that don't have low-level information in the database.
+
+    Args:
+        dataset: a dataset loaded with get_dataset
+
+    Returns:
+        Dataset filtered content (same structure as args dataset)
+        Dictionary of class name and corresponding list of ignored recordings
+    """
+    rec_memo = {}
+    ignored_recordings = defaultdict(list)
+
+    for cls in dataset["classes"]:
+        for i, recording_mbid in enumerate(cls["recordings"]):
+            if recording_mbid in rec_memo and rec_memo[recording_mbid]:
+                pass
+            if db.data.count_lowlevel(recording_mbid) > 0:
```

Comment:
(we talked about this on irc)
For large datasets (e.g. hundreds or thousands of recordings) a check like this will take a long time, especially if the dataset is valid (and it has to check _every_ recording). I'd prefer to try something like a join of the dataset table against the lowlevel table on the mbid field to get all missing recordings at once